[BUG]

### Python 版本

3.10

### SuggarChat 版本

3.2.2

### 发生了什么?

在nonebot配置文件中配置SuperUsers时如果配置了纯字符型管理员（如： SUPERUSERS=["123456", "GOLU"]）,会在图中位置引发类型转换异常 [nonebot_plugin_suggarchat/check_rule.py]：

<img width="592" height="125" alt="Image" src="https://github.com/user-attachments/assets/7b49e64e-cebf-4707-bba8-3dfda5406095" />

### 如何复现

nonebot配置文件中配置SuperUsers时配置纯文本类型管理员

### 期望的结果?

应当捕获此异常并提醒或忽略文本型管理员，或是支持字符型管理员，部分平台账号不是纯文本

### SuggarChat 运行日志

```shell
08-11 16:00:22 [ERROR] nonebot | Rule check failed for Matcher(type='message', module=nonebot_plugin_suggarchat.matcher_manager, lineno=36).
Traceback (most recent call last):
  File "<string>", line 37, in <module>
  File "/home/nonebot/golu-bot/.venv/lib/python3.10/site-packages/nonebot/__init__.py", line 337, in run
    get_driver().run(*args, **kwargs)
  File "/home/nonebot/golu-bot/.venv/lib/python3.10/site-packages/nonebot/drivers/fastapi.py", line 187, in run
    uvicorn.run(
  File "/home/nonebot/golu-bot/.venv/lib/python3.10/site-packages/uvicorn/main.py", line 569, in run
    server.run()
  File "/home/nonebot/golu-bot/.venv/lib/python3.10/site-packages/uvicorn/server.py", line 60, in run
    return asyncio.run(self.serve(sockets=sockets))
  File "/usr/local/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/home/nonebot/golu-bot/.venv/lib/python3.10/site-packages/nonebot/utils.py", line 259, in run_coro_with_shield
    return await coro
  File "/home/nonebot/golu-bot/.venv/lib/python3.10/site-packages/nonebot/message.py", line 498, in check_and_run_matcher
    if not await _check_matcher(
> File "/home/nonebot/golu-bot/.venv/lib/python3.10/site-packages/nonebot/message.py", line 402, in _check_matcher
    if not await Matcher.check_rule(bot, event, state, stack, dependency_cache):
  File "/home/nonebot/golu-bot/.venv/lib/python3.10/site-packages/nonebot/internal/matcher/matcher.py", line 405, in check_rule
    return event_type == (cls.type or event_type) and await cls.rule(
  File "/home/nonebot/golu-bot/.venv/lib/python3.10/site-packages/nonebot/internal/rule.py", line 97, in __call__
    with catch({SkippedException: _handle_skipped_exception}):
  File "/home/nonebot/golu-bot/.venv/lib/python3.10/site-packages/exceptiongroup/_catch.py", line 39, in __exit__
    raise unhandled from exc.__cause__
  File "/home/nonebot/golu-bot/.venv/lib/python3.10/site-packages/nonebot/internal/rule.py", line 98, in __call__
    async with anyio.create_task_group() as tg:
  File "/home/nonebot/golu-bot/.venv/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 772, in __aexit__
    raise BaseExceptionGroup(
exceptiongroup.ExceptionGroup: unhandled errors in a TaskGroup (1 sub-exception)
```

--- a/nonebot_plugin_suggarchat/builtin_hook.py
+++ b/nonebot_plugin_suggarchat/builtin_hook.py
@@ -24,6 +24,8 @@ from .utils.libchat import (
 from .utils.llm_tools.builtin_tools import REPORT_TOOL, report
 from .utils.llm_tools.manager import ToolsManager
 from .utils.memory import (
+    Message,
+    ToolResult,
     get_memory_data,
     write_memory_data,
 )
@@ -56,14 +58,14 @@ async def tools_callerdler(event: BeforeChatEvent) -> None:
             tools.extend(ToolsManager().tools_meta_dict(exclude_none=True).values())
             response_msg = await tools_caller(
                 [
-                    *deepcopy([i for i in msg_list if i["role"] == "system"]),
-                    deepcopy(msg_list)[-1],
+                    *deepcopy([i.model_dump() for i in msg_list if i.role == "system"]),
+                    deepcopy(msg_list)[-1].model_dump(),
                 ],
                 tools,
             )
             tool_calls = response_msg.tool_calls
             if tool_calls:
-                msg_list.append(dict(response_msg))
+                msg_list.append(Message.model_validate(dict(response_msg)))
                 for tool_call in tool_calls:
                     function_name = tool_call.function.name
                     function_args: dict = json.loads(tool_call.function.arguments)
@@ -98,12 +100,12 @@ async def tools_callerdler(event: BeforeChatEvent) -> None:
                                 )
                                 continue
                     logger.debug(f"函数{function_name}返回：{func_response}")
-                    msg = {
-                        "tool_call_id": tool_call.id,
-                        "role": "tool",
-                        "name": function_name,
-                        "content": func_response,
-                    }
+
+                    msg = ToolResult(
+                        content=func_response,
+                        name=function_name,
+                        tool_call_id=tool_call.id,
+                    )
                     msg_list.append(msg)
         except Exception as e:
             if isinstance(e, ChatException):

--- a/nonebot_plugin_suggarchat/check_rule.py
+++ b/nonebot_plugin_suggarchat/check_rule.py
@@ -13,7 +13,7 @@ from .utils.functions import (
     get_current_datetime_timestamp,
     synthesize_message,
 )
-from .utils.memory import get_memory_data, write_memory_data
+from .utils.memory import Message, get_memory_data, write_memory_data
 
 nb_config = get_driver().config
 
@@ -43,7 +43,7 @@ async def is_group_admin(event: GroupMessageEvent, bot: Bot) -> bool:
 
 async def is_bot_admin(event: MessageEvent, bot: Bot) -> bool:
     return event.user_id in config_manager.config.admin.admins + [
-        int(user) for user in nb_config.superusers
+        int(user) for user in nb_config.superusers if user.isdigit()
     ]
 
 
@@ -123,28 +123,26 @@ async def should_respond_to_message(event: MessageEvent, bot: Bot) -> bool:
 
         # 生成消息内容并记录到内存
         content_message = f"[{role}][{Date}][{user_name}（{user_id}）]说:{content}"
-        fwd_msg = {"role": "user", "content": "<FORWARD_MSG>\n" + content_message}
+        fwd_msg = Message(role="user", content="<FORWARD_MSG>\n" + content_message)
         message_l = memory_data.memory.messages
         if not message_l:
             message_l.append(fwd_msg)
-        elif (
-            not isinstance(message_l[-1].get("content"), str)
-            or message_l[-1]["role"] != "user"
-        ):
+        elif not isinstance(message_l[-1].content, str) or message_l[-1].role != "user":
             message_l.append(fwd_msg)
-        elif not message_l[-1]["content"].startswith("<FORWARD_MSG>"):
+        elif not message_l[-1].content.startswith("<FORWARD_MSG>"):
             message_l.append(fwd_msg)
         else:
-            message_l[-1]["content"] += "\n" + content_message
-        if (
-            len(message_l[-1]["content"])
-            > config_manager.config.llm_config.memory_lenth_limit * 10
+            message_l[-1].content += "\n" + content_message
+        if len(
+            message_l[-1].content
+        ) > config_manager.config.llm_config.memory_lenth_limit * 10 and isinstance(
+            message_l[-1].content, str
         ):
-            lines = message_l[-1]["content"].splitlines(keepends=True)
+            lines = message_l[-1].content.splitlines(keepends=True)
             if len(lines) >= 2:
                 # 删除索引为1的第二行
                 del lines[1]
-            message_l[-1]["content"] = "".join(lines)
+            message_l[-1].content = "".join(lines)
         memory_data.memory.messages = message_l
 
         # 写入内存数据

--- a/nonebot_plugin_suggarchat/event.py
+++ b/nonebot_plugin_suggarchat/event.py
@@ -8,6 +8,8 @@ from nonebot.adapters.onebot.v11 import (
 )
 from typing_extensions import override
 
+from .utils.memory import Message, ToolResult
+
 
 class EventTypeEnum(str, Enum):
     """
@@ -127,7 +129,7 @@ class SuggarEvent(BasicEvent):
         model_response: list[str],
         nbevent: Event,
         user_id: int,
-        send_message: list[str],
+        send_message: list[Message | ToolResult],
     ):
         """
         初始化SuggarEvent对象
@@ -142,13 +144,11 @@ class SuggarEvent(BasicEvent):
         # 保存NoneBot事件对象
         self._nbevent = nbevent
         # 初始化模型响应文本
-        self._modelResponse: list = model_response
-        # 这里使用列表是因为Python的底层指针允许列表对象引用同一块内存，所以可以修改列表中的元素，而不需要重新分配内存，这样就能实现修改模型的响应了。
-
+        self._modelResponse: list[str] = model_response
         # 初始化用户ID
         self._user_id: int = user_id
         # 初始化要发送的消息内容
-        self._send_message: list = send_message
+        self._send_message: list[Message | ToolResult] = send_message
 
     def __bool__(self):
         return True
@@ -267,7 +267,7 @@ class ChatEvent(SuggarEvent):
     def __init__(
         self,
         nbevent: MessageEvent,
-        send_message: list[str],
+        send_message: list[Message | ToolResult],
         model_response: list[str],
         user_id: int,
     ):

--- a/nonebot_plugin_suggarchat/handlers/chat.py
+++ b/nonebot_plugin_suggarchat/handlers/chat.py
@@ -39,7 +39,14 @@ from ..utils.functions import (
 )
 from ..utils.libchat import get_chat
 from ..utils.lock import get_group_lock, get_private_lock
-from ..utils.memory import MemoryModel, get_memory_data, write_memory_data
+from ..utils.memory import (
+    Memory,
+    MemoryModel,
+    Message,
+    ToolResult,
+    get_memory_data,
+    write_memory_data,
+)
 from ..utils.tokenizer import hybrid_token_count
 
 command_prefix = get_driver().config.command_start or "/"
@@ -130,10 +137,7 @@ async def chat(event: MessageEvent, matcher: Matcher, bot: Bot):
             text = event.message.extract_plain_text()
 
         group_data.memory.messages.append(
-            {
-                "role": "user",
-                "content": text,
-            }
+            Message.model_validate({"role": "user", "content": text})
         )
         if chat_manager.debug:
             logger.debug(f"当前群组提示词：\n{config_manager.group_train}")
@@ -151,10 +155,10 @@ async def chat(event: MessageEvent, matcher: Matcher, bot: Bot):
 
         # 记录模型回复
         group_data.memory.messages.append(
-            {
-                "role": "assistant",
-                "content": str(response),
-            }
+            Message(
+                role="assistant",
+                content=response,
+            )
         )
         await send_response(event, response)
 
@@ -212,7 +216,7 @@ async def chat(event: MessageEvent, matcher: Matcher, bot: Bot):
                     },
                 ]
                 + [
-                    {"type": "input_image", "url": seg.data.get("url")}
+                    {"type": "image_url", "image_url": {"url": seg.data.get("url")}}
                     for seg in event.message
                     if seg.data.get("type") == "image"
                 ]
@@ -221,7 +225,9 @@ async def chat(event: MessageEvent, matcher: Matcher, bot: Bot):
             )
         else:
             text = event.message.extract_plain_text()
-        private_data.memory.messages.append({"role": "user", "content": text})
+        private_data.memory.messages.append(
+            Message.model_validate({"role": "user", "content": text})
+        )
         if chat_manager.debug:
             logger.debug(f"当前私聊提示词：\n{config_manager.private_train}")
         # 控制记忆长度和 token 限制
@@ -238,10 +244,10 @@ async def chat(event: MessageEvent, matcher: Matcher, bot: Bot):
 
         # 记录模型回复
         private_data.memory.messages.append(
-            {
-                "role": "assistant",
-                "content": str(response),
-            }
+            Message(
+                content=response,
+                role="assistant",
+            )
         )
         await send_response(event, response)
 
@@ -276,10 +282,7 @@ async def chat(event: MessageEvent, matcher: Matcher, bot: Bot):
                     float(config_manager.config.session.session_control_time * 60)
                 ):
                     data.sessions.append(
-                        {
-                            "messages": data.memory.messages,
-                            "time": time_now,
-                        }
+                        Memory(messages=data.memory.messages, time=time_now)
                     )
                     while (
                         len(data.sessions)
@@ -313,7 +316,7 @@ async def chat(event: MessageEvent, matcher: Matcher, bot: Bot):
 
                     del session_clear_map[session_id]
 
-                    data.memory.messages = data.sessions[-1]["messages"]
+                    data.memory.messages = data.sessions[-1].messages
                     data.sessions.pop()
                     await matcher.send("让我们继续聊天吧～")
                     await write_memory_data(event, data)
@@ -362,21 +365,21 @@ async def chat(event: MessageEvent, matcher: Matcher, bot: Bot):
         # Process multimodal messages when needed
         for message in data.memory.messages:
             if (
-                isinstance(message["content"], dict)
+                isinstance(message.content, list)
                 and not is_multimodal
-                and message["role"] == "user"
+                and message.role == "user"
             ):
                 message_text = ""
-                for content_part in message["content"]:
-                    if content_part["type"] == "text":
-                        message_text += content_part["text"]
-                message["content"] = message_text
+                for content_part in message.content:
+                    if content_part.type == "text":
+                        message_text += content_part.text
+                message.content = message_text
 
         # Enforce memory length limit
         while len(data.memory.messages) > 0:
             if (
                 len(data.memory.messages) > memory_length_limit
-                or data.memory.messages[0]["role"] != "user"
+                or data.memory.messages[0].role != "user"
             ):
                 del data.memory.messages[0]
             else:
@@ -387,7 +390,7 @@ async def chat(event: MessageEvent, matcher: Matcher, bot: Bot):
         控制 token 数量，删除超出限制的旧消息，返回处理后的Tokens。
         """
         train = copy.deepcopy(train)
-        memory_l = [train, *copy.deepcopy(data.memory.messages.copy())]  # type: list[dict]
+        memory_l = [train, *[i.model_dump() for i in data.memory.messages]]
         full_string = ""
         for st in memory_l:
             if isinstance(st["content"], str):
@@ -451,11 +454,11 @@ async def chat(event: MessageEvent, matcher: Matcher, bot: Bot):
             f"\n以下是一些补充内容，如果与上面任何一条有冲突请忽略。\n{data.prompt if data.prompt != '' else '无'}"
         )
         send_messages = copy.deepcopy(data.memory.messages)
-        send_messages.insert(0, train)
+        send_messages.insert(0, Message.model_validate(train))
         return send_messages
 
     async def process_chat(
-        event: MessageEvent, send_messages: list, tokens: int
+        event: MessageEvent, send_messages: list[Message | ToolResult], tokens: int
     ) -> str:
         """
         调用聊天模型生成回复，并触发相关事件。

--- a/nonebot_plugin_suggarchat/handlers/del_memory.py
+++ b/nonebot_plugin_suggarchat/handlers/del_memory.py
@@ -3,7 +3,7 @@ from nonebot.adapters.onebot.v11 import Bot, MessageEvent
 from nonebot.matcher import Matcher
 
 from ..check_rule import is_group_admin_if_is_in_group
-from ..utils.memory import get_memory_data, write_memory_data
+from ..utils.memory import get_memory_data
 
 
 async def del_memory(bot: Bot, event: MessageEvent, matcher: Matcher):
@@ -12,7 +12,7 @@ async def del_memory(bot: Bot, event: MessageEvent, matcher: Matcher):
         return
     data = await get_memory_data(event)
     data.memory.messages.clear()
-    await write_memory_data(event, data)
+    await data.save(event)
     await matcher.send("上下文已清除")
     logger.info(
         f"{event.get_event_name()}:{getattr(event, 'group_id') if hasattr(event, 'group_id') else event.user_id} 的记忆已清除"

--- a/nonebot_plugin_suggarchat/handlers/disable.py
+++ b/nonebot_plugin_suggarchat/handlers/disable.py
@@ -3,7 +3,7 @@ from nonebot.adapters.onebot.v11 import Bot, GroupMessageEvent
 from nonebot.matcher import Matcher
 
 from ..check_rule import is_bot_admin
-from ..utils.memory import get_memory_data, write_memory_data
+from ..utils.memory import get_memory_data
 
 
 async def disable(bot: Bot, event: GroupMessageEvent, matcher: Matcher):
@@ -15,10 +15,8 @@ async def disable(bot: Bot, event: GroupMessageEvent, matcher: Matcher):
 
     # 获取并更新群聊状态数据
     data = await get_memory_data(event)
-    if data["id"] == event.group_id:
-        if data["enable"]:
-            data["enable"] = False
-        await matcher.send("聊天功能已禁用")
+    data.enable = False
+    await data.save(event)
+    await matcher.send("聊天功能已禁用")
 
     # 保存更新后的群聊状态数据
-    await write_memory_data(event, data)

--- a/nonebot_plugin_suggarchat/handlers/enable.py
+++ b/nonebot_plugin_suggarchat/handlers/enable.py
@@ -3,7 +3,7 @@ from nonebot.adapters.onebot.v11 import Bot, GroupMessageEvent
 from nonebot.matcher import Matcher
 
 from ..check_rule import is_group_admin
-from ..utils.memory import get_memory_data, write_memory_data
+from ..utils.memory import get_memory_data
 
 
 async def enable(bot: Bot, event: GroupMessageEvent, matcher: Matcher):
@@ -15,10 +15,6 @@ async def enable(bot: Bot, event: GroupMessageEvent, matcher: Matcher):
     # 获取当前群组的记忆数据
     data = await get_memory_data(event)
     # 检查记忆数据是否与当前群组匹配
-    if data["id"] == event.group_id:
-        # 如果聊天功能未启用，则启用并发送提示
-        if not data["enable"]:
-            data["enable"] = True
-        await matcher.send("聊天启用")
-    # 更新记忆数据
-    await write_memory_data(event, data)
+    data.enable = True
+    await data.save(event)
+    await matcher.send("已启用聊天功能")

--- a/nonebot_plugin_suggarchat/handlers/prompt.py
+++ b/nonebot_plugin_suggarchat/handlers/prompt.py
@@ -6,7 +6,7 @@ from ..check_rule import (
     is_group_admin_if_is_in_group,
 )
 from ..config import config_manager
-from ..utils.memory import get_memory_data, write_memory_data
+from ..utils.memory import get_memory_data
 
 
 async def prompt(
@@ -56,4 +56,4 @@ async def prompt(
         return
 
     # 更新记忆数据
-    await write_memory_data(event, data)
+    await data.save(event)

--- a/nonebot_plugin_suggarchat/handlers/sessions.py
+++ b/nonebot_plugin_suggarchat/handlers/sessions.py
@@ -1,4 +1,5 @@
 import time
+from copy import deepcopy
 from datetime import datetime
 
 from nonebot.adapters.onebot.v11 import Bot, Message, MessageEvent
@@ -34,7 +35,7 @@ async def sessions(
         """将当前会话覆盖为指定编号的会话"""
         try:
             if len(arg_list) >= 2:
-                data.memory.messages = data.sessions[int(arg_list[1])].messages
+                data.memory.messages = deepcopy(data.sessions[int(arg_list[1])].messages)
                 data.timestamp = time.time()
                 await write_memory_data(event, data)
                 await matcher.send("完成记忆覆盖。")

--- a/nonebot_plugin_suggarchat/handlers/sessions.py
+++ b/nonebot_plugin_suggarchat/handlers/sessions.py
@@ -8,7 +8,7 @@ from nonebot.params import CommandArg
 
 from ..check_rule import is_group_admin_if_is_in_group
 from ..config import config_manager
-from ..utils.memory import MemoryModel, get_memory_data, write_memory_data
+from ..utils.memory import Memory, MemoryModel, get_memory_data, write_memory_data
 
 
 async def sessions(
@@ -24,7 +24,8 @@ async def sessions(
             await matcher.finish("没有历史会话")
         message_content = "历史会话\n"
         for index, msg in enumerate(data.sessions):
-            message_content += f"编号：{index}) ：{msg['messages'][0]['content'][9:]}... 时间：{datetime.fromtimestamp(msg['time']).strftime('%Y-%m-%d %I:%M:%S %p')}\n"
+            if msg.messages:
+                message_content += f"编号：{index}) ：{msg.messages[0].content[63:68]}... 时间：{datetime.fromtimestamp(msg.time).strftime('%Y-%m-%d %I:%M:%S %p')}\n"
         await matcher.finish(message_content)
 
     async def set_session(
@@ -33,7 +34,7 @@ async def sessions(
         """将当前会话覆盖为指定编号的会话"""
         try:
             if len(arg_list) >= 2:
-                data.memory.messages = data.sessions[int(arg_list[1])]["messages"]
+                data.memory.messages = data.sessions[int(arg_list[1])].messages
                 data.timestamp = time.time()
                 await write_memory_data(event, data)
                 await matcher.send("完成记忆覆盖。")
@@ -64,10 +65,7 @@ async def sessions(
         try:
             if data.memory.messages:
                 data.sessions.append(
-                    {
-                        "messages": data.memory.messages,
-                        "time": time.time(),
-                    }
+                    Memory(messages=data.memory.messages, time=time.time())
                 )
                 data.memory.messages = []
                 data.timestamp = time.time()

--- a/nonebot_plugin_suggarchat/handlers/sessions.py
+++ b/nonebot_plugin_suggarchat/handlers/sessions.py
@@ -105,18 +105,18 @@ async def sessions(
         await display_sessions(data)
 
     # 根据命令执行对应操作
-    command = arg_list[0]
-    if command == "set":
-        await set_session(data, arg_list, event)
-    elif command == "del":
-        await delete_session(data, arg_list, event)
-    elif command == "archive":
-        await archive_session(data, event)
-    elif command == "clear":
-        await clear_sessions(data, event)
-    elif command == "help":
-        await matcher.finish(
-            "Sessions指令帮助：\nset：覆盖当前会话为指定编号的会话\ndel：删除指定编号的会话\narchive：归档当前会话\nclear：清空所有会话\n"
-        )
-    else:
-        await matcher.finish("未知命令，请输入/help查看帮助。")
+    match arg_list[0]:
+        case "set":
+            await set_session(data, arg_list, event)
+        case "del":
+            await delete_session(data, arg_list, event)
+        case "archive":
+            await archive_session(data, event)
+        case "clear":
+            await clear_sessions(data, event)
+        case "help":
+            await matcher.finish(
+                "Sessions指令帮助：\nset：覆盖当前会话为指定编号的会话\ndel：删除指定编号的会话\narchive：归档当前会话\nclear：清空所有会话\n"
+            )
+        case _:
+            await matcher.finish("未知命令，请输入/help查看帮助。")

--- a/nonebot_plugin_suggarchat/utils/functions.py
+++ b/nonebot_plugin_suggarchat/utils/functions.py
@@ -189,13 +189,13 @@ async def synthesize_forward_message(forward_msg: dict, bot: Bot) -> str:
                 result += f"{segment['data']['content']}"
             elif isinstance(segment["data"]["content"], list):
                 for segments in segment["data"]["content"]:
-                    segments_type = segments["type"]
-                    if segments_type == "text":
-                        result += f"{segments['data']['text']}"
-                    elif segments_type == "at":
-                        result += f" [@{segments['data']['qq']}]"
-                    elif segments_type == "forward":
-                        result += f"\\（合并转发:{await synthesize_forward_message(await bot.get_forward_msg(id=segments['data']['id']), bot)}）\\"
+                    match segments["type"]:
+                        case "text":
+                            result += f"{segments['data']['text']}"
+                        case "at":
+                            result += f" [@{segments['data']['qq']}]"
+                        case "forward":
+                            result += f"\\（合并转发:{await synthesize_forward_message(await bot.get_forward_msg(id=segments['data']['id']), bot)}）\\"
         except Exception as e:
             logger.opt(colors=True, exception=e).error(f"合成转发消息时出错：{e!s}'")
             result += "<!--该消息段无法被解析-->"

--- a/nonebot_plugin_suggarchat/utils/functions.py
+++ b/nonebot_plugin_suggarchat/utils/functions.py
@@ -176,25 +176,29 @@ async def synthesize_forward_message(forward_msg: dict, bot: Bot) -> str:
     """
     result = ""
     for segment in forward_msg:
-        if isinstance(segment["data"], str):
-            try:
-                segment["data"] = json.loads(segment["data"])
-            except Exception:
-                result += segment["data"] + "<!--该消息段无法被解析-->"
-        nickname: str = segment["data"]["nickname"]
-        qq: str = segment["data"]["user_id"]
-        result += f"[{nickname}({qq})]说："
-        if isinstance(segment["data"]["content"], str):
-            result += f"{segment['data']['content']}"
-        elif isinstance(segment["data"]["content"], list):
-            for segments in segment["data"]["content"]:
-                segments_type = segments["type"]
-                if segments_type == "text":
-                    result += f"{segments['data']['text']}"
-                elif segments_type == "at":
-                    result += f" [@{segments['data']['qq']}]"
-                elif segments_type == "forward":
-                    result += f"\\（合并转发:{await synthesize_forward_message(await bot.get_forward_msg(id=segments['data']['id']), bot)}）\\"
+        try:
+            if isinstance(segment["data"], str):
+                try:
+                    segment["data"] = json.loads(segment["data"])
+                except Exception:
+                    result += segment["data"] + "<!--该消息段无法被解析-->"
+            nickname: str = segment["data"]["nickname"]
+            qq: str = segment["data"]["user_id"]
+            result += f"[{nickname}({qq})]说："
+            if isinstance(segment["data"]["content"], str):
+                result += f"{segment['data']['content']}"
+            elif isinstance(segment["data"]["content"], list):
+                for segments in segment["data"]["content"]:
+                    segments_type = segments["type"]
+                    if segments_type == "text":
+                        result += f"{segments['data']['text']}"
+                    elif segments_type == "at":
+                        result += f" [@{segments['data']['qq']}]"
+                    elif segments_type == "forward":
+                        result += f"\\（合并转发:{await synthesize_forward_message(await bot.get_forward_msg(id=segments['data']['id']), bot)}）\\"
+        except Exception as e:
+            logger.opt(colors=True, exception=e).error(f"合成转发消息时出错：{e!s}'")
+            result += "<!--该消息段无法被解析-->"
         result += "\n"
     return result
 

--- a/nonebot_plugin_suggarchat/utils/libchat.py
+++ b/nonebot_plugin_suggarchat/utils/libchat.py
@@ -16,11 +16,12 @@ from openai.types.chat.chat_completion_tool_choice_option_param import (
 from ..chatmanager import chat_manager
 from ..config import config_manager
 from .functions import remove_think_tag
+from .memory import Message, ToolResult
 from .protocol import AdapterManager, ModelAdapter
 
 
 async def tools_caller(
-    messages: list,
+    messages: Iterable,
     tools: list,
     tool_choice: ChatCompletionToolChoiceOptionParam | None = None,
 ) -> ChatCompletionMessage:
@@ -75,7 +76,7 @@ async def tools_caller(
 
 
 async def get_chat(
-    messages: list,
+    messages: list[Message | ToolResult],
     tokens: int = 0,
 ) -> str:
     """获取聊天响应"""
@@ -107,7 +108,9 @@ async def get_chat(
                 ex = None
                 try:
                     processer = adapter(preset, config_manager.config)
-                    response = await processer.call_api(messages)
+                    response = await processer.call_api(
+                        [i.model_dump() for i in messages]
+                    )
                     break  # 成功获取响应，跳出重试循环
                 except Exception as e:
                     ex = e

--- a/nonebot_plugin_suggarchat/utils/memory.py
+++ b/nonebot_plugin_suggarchat/utils/memory.py
@@ -1,7 +1,9 @@
+from __future__ import annotations
+
 import json
 import time
 from pathlib import Path
-from typing import Any
+from typing import Any, Literal
 
 import aiofiles
 from nonebot import logger
@@ -11,7 +13,8 @@ from nonebot.adapters.onebot.v11 import (
     PokeNotifyEvent,
     PrivateMessageEvent,
 )
-from pydantic import BaseModel, Field
+from pydantic import BaseModel as Model
+from pydantic import Field
 
 from ..chatmanager import chat_manager
 from ..config import config_manager
@@ -19,20 +22,7 @@ from .functions import convert_to_utf8
 from .lock import rw_lock
 
 
-class Memory(BaseModel):
-    messages: list[dict[str, Any]] = Field(default_factory=list)
-
-
-class MemoryModel(BaseModel, extra="allow"):
-    id: int = Field(..., description="ID")
-    enable: bool = Field(default=True, description="是否启用")
-    memory: Memory = Field(default=Memory(), description="记忆")
-    full: bool = Field(default=False, description="是否启用Fullmode")
-    sessions: list[dict[str, Any]] = Field(default=[], description="会话")
-    timestamp: float = Field(default=time.time(), description="时间戳")
-    fake_people: bool = Field(default=False, description="是否启用假人")
-    prompt: str = Field(default="", description="用户自定义提示词")
-
+class BaseModel(Model):
     def __str__(self) -> str:
         return json.dumps(self.model_dump(), ensure_ascii=True)
 
@@ -44,6 +34,51 @@ class MemoryModel(BaseModel, extra="allow"):
 
     def __setitem__(self, key: str, value: Any) -> None:
         self.__setattr__(key, value)
+
+
+class ImageUrl(BaseModel):
+    url: str = Field(..., description="图片URL")
+
+
+class ImageContent(BaseModel):
+    type: Literal["image_url"] = "image_url"
+    image_url: ImageUrl = Field(..., description="图片URL")
+
+
+class TextContent(BaseModel):
+    type: Literal["text"] = "text"
+    text: str = Field(..., description="文本内容")
+
+
+class Message(BaseModel):
+    role: Literal["user", "assistant", "system"] = Field(..., description="角色")
+    content: str | list[TextContent | ImageContent] = Field(..., description="内容")
+
+
+class ToolResult(BaseModel):
+    role: Literal["tool"] = Field(default="tool", description="角色")
+    name: str = Field(..., description="工具名称")
+    content: str = Field(..., description="工具返回内容")
+    tool_call_id: str = Field(..., description="工具调用ID")
+
+
+class Memory(BaseModel):
+    messages: list[Message | ToolResult] = Field(default_factory=list)
+    time: float = Field(default_factory=time.time, description="时间戳")
+
+
+class MemoryModel(BaseModel, extra="allow"):
+    enable: bool = Field(default=True, description="是否启用")
+    memory: Memory = Field(default=Memory(), description="记忆")
+    full: bool = Field(default=False, description="是否启用Fullmode")
+    sessions: list[Memory] = Field(default=[], description="会话")
+    timestamp: float = Field(default=time.time(), description="时间戳")
+    fake_people: bool = Field(default=False, description="是否启用假人")
+    prompt: str = Field(default="", description="用户自定义提示词")
+
+    async def save(self, event: Event) -> None:
+        """保存当前内存数据到文件"""
+        await write_memory_data(event, self)
 
 
 async def get_memory_data(event: Event) -> MemoryModel:
@@ -74,7 +109,7 @@ async def get_memory_data(event: Event) -> MemoryModel:
                     str(conf_path),
                     "w",
                 ) as f:
-                    await f.write(str(MemoryModel(id=group_id)))
+                    await f.write(MemoryModel().model_dump_json())
         elif (
             not isinstance(event, PrivateMessageEvent)
             and isinstance(event, PokeNotifyEvent)
@@ -86,7 +121,7 @@ async def get_memory_data(event: Event) -> MemoryModel:
                     str(conf_path),
                     "w",
                 ) as f:
-                    await f.write(str(MemoryModel(id=user_id)))
+                    await f.write(MemoryModel().model_dump_json())
         assert conf_path is not None, "conf_path is None"
         convert_to_utf8(conf_path)
         async with aiofiles.open(
@@ -122,13 +157,7 @@ async def write_memory_data(event: Event, data: MemoryModel) -> None:
                         str(conf_path),
                         "w",
                     ) as f:
-                        await f.write(
-                            str(
-                                MemoryModel(
-                                    id=group_id,
-                                )
-                            )
-                        )
+                        await f.write(MemoryModel().model_dump_json())
             else:
                 user_id = event.user_id
                 conf_path = Path(private_memory / f"{user_id}.json")
@@ -137,13 +166,7 @@ async def write_memory_data(event: Event, data: MemoryModel) -> None:
                         str(conf_path),
                         "w",
                     ) as f:
-                        await f.write(
-                            str(
-                                MemoryModel(
-                                    id=user_id,
-                                )
-                            )
-                        )
+                        await f.write(MemoryModel().model_dump_json())
         assert conf_path is not None
         async with aiofiles.open(
             str(conf_path),

--- a/nonebot_plugin_suggarchat/utils/memory.py
+++ b/nonebot_plugin_suggarchat/utils/memory.py
@@ -71,7 +71,7 @@ class MemoryModel(BaseModel, extra="allow"):
     enable: bool = Field(default=True, description="是否启用")
     memory: Memory = Field(default=Memory(), description="记忆")
     full: bool = Field(default=False, description="是否启用Fullmode")
-    sessions: list[Memory] = Field(default=[], description="会话")
+    sessions: list[Memory] = Field(default_factory=list, description="会话")
     timestamp: float = Field(default=time.time(), description="时间戳")
     fake_people: bool = Field(default=False, description="是否启用假人")
     prompt: str = Field(default="", description="用户自定义提示词")

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,6 +1,6 @@
 [project]
 name = "nonebot_plugin_suggarchat"
-version = "3.2.2"
+version = "3.3.0.dev1"
 description = "SuggarChat chat framework"
 authors = [{ name = "LiteSuggarDEV", email = "windowserror@163.com" }]
 dependencies = [

